File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Armenia.m
//
//  UIImage+Armenia.m
//  CountriesFlags
//

#import "Countries/UIImage+Armenia.h"

@implementation UIImage (Armenia)

+ (UIImage * _Nullable)flagArmeniaWithSize:(CGSize)size {
    UIColor* colorBlue = [UIColor colorWithRed:0.0f
                                         green:51.0f/255.0f
                                          blue:160.0f/255.0f
                                         alpha:1.0f];
    UIColor* colorRed = [UIColor colorWithRed:217.0f/255.0f
                                        green:0.0f
                                         blue:18.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorOrange = [UIColor colorWithRed:245.0f/255.0f
                                           green:168.0f/255.0f
                                            blue:0.0f
                                           alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f, size.width, size.height / 3.0f));
    
    [colorOrange setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f * 2.0f, size.width, size.height / 3.0f));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Armenia.swift
//
//  UIImage+Armenia.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Armenia "wikipedia"
     
     Generates the flag of Armenia.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Armenia.
     */
    internal static func flagArmenia(with size: CGSize) -> UIImage? {
        let colorBlue = UIColor(red: 0.0,
                                green: 51.0 / 255.0,
                                blue: 160.0 / 255.0,
                                alpha: 1.0)
        let colorRed = UIColor(red: 217.0 / 255.0,
                               green: 0.0,
                               blue: 18.0 / 255.0,
                               alpha: 1.0)
        let colorOrange = UIColor(red: 245.0 / 255.0,
                                  green: 168.0 / 255.0,
                                  blue: 0.0,
                                  alpha: 1.0)
        let lineSize = CGSize(width: size.width, height: size.height / 3.0)

        UIGraphicsBeginImageContextWithOptions(size, true, 0.0)

        colorRed.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorBlue.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 3.0), size: lineSize))

        colorOrange.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 3.0 * 2.0), size: lineSize))

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
